packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -274,6 +227,23 @@ const BlockInspectorSingleBlock = ( { clientId, blockName } ) => {
 	);
 	const blockInformation = useBlockDisplayInformation( clientId );
 	const borderPanelLabel = useBorderPanelLabel( { blockName } );
+	const contentClientIds = useSelect(
+		( select ) => {
+			const {
+				getClientIdsOfDescendants,
+				getBlockName,
+				getBlockEditingMode,
+			} = select( blockEditorStore );
+			return isContentLockingParent
+				? getClientIdsOfDescendants( clientId ).filter(
+						( current ) =>
+							getBlockName( current ) !== 'core/list-item' &&
+							getBlockEditingMode( current ) === 'contentOnly'
+				  )
+				: undefined;
+		},
+		[ isContentLockingParent, clientId ]
+	);
```

Comment:
A micro-optimization, but we can return early when `! isContentLockingParent` and avoid creating a store subscription.

Reply:
I've updated the code. Can you confirm it's as you expected?